pkg/build/builder/cmd/builder.go

```diff
@@ -67,6 +77,59 @@ func run(builderFactory factoryFunc) {
 
 }
 
+func setupSourceSecret(sourceSecretName string) (err error) {
+	// TODO: remove this when permissions for secrets will be fixed upstream
+	// this is to restrict access permissions to very low acceptable level
+	secretTmpDir, err := ioutil.TempDir("", "tmpsecret")
+	if err != nil {
+		return
+	}
+	cmd := exec.Command("cp", "-R", ".", secretTmpDir)
+	cmd.Dir = os.Getenv("SOURCE_SECRET_PATH")
+	if err = cmd.Run(); err != nil {
+		return
+	}
+	secretFiles, err := ioutil.ReadDir(secretTmpDir)
+	if err != nil {
+		return
+	}
+	for _, file := range secretFiles {
+		if err = os.Chmod(filepath.Join(secretTmpDir, file.Name()), 0600); err != nil {
+			return
+		}
+	}
+	os.Setenv("SOURCE_SECRET_PATH", secretTmpDir)
+	// TODO: end
+
+	sourceSecretDir := os.Getenv("SOURCE_SECRET_PATH")
+	files, err := ioutil.ReadDir(sourceSecretDir)
+	if err != nil {
+		return
+	}
+	found := false
+
+SCMAuthLoop:
+	for _, scmAuth := range scmauth.SCMAuthsOrder {
+		glog.V(3).Infof("Checking for '%s' in secret '%s'", scmAuth, sourceSecretName)
+		for _, file := range files {
+			if file.Name() == scmAuth {
+				glog.Infof("Using '%s' from secret '%s'", scmAuth, sourceSecretName)
+				if err := scmauth.SCMAuths[scmAuth].Setup(sourceSecretDir); err != nil {
+					glog.Warningf("Error setting up '%s': %v", scmAuth, err)
```

Comment:
@soltysh if I understand this properly, then this code will identify the file name as the one that provide the 'secret' mechanism... So this code relies on the fact that the file name maps to the 'authentication' mechanism... Is there any use-case where you have file named 'foo' that will be used in more than one authentication methods?